sccm-currentbranch/prereqs/prereq.azuredeploy.json

```diff
@@ -0,0 +1,266 @@
+      "subnetRef": "[concat(variables('vnetId'), '/subnets/',variables('existingSubnetName'))]",
+      "privateIPAllocationMethod": "Static"
+    },
+    "securityGroupRule": {
+      "name": "default-allow-rdp",
+      "priority": 1000,
+      "sourceAddressPrefix": "*",
+      "protocol": "Tcp",
+      "destinationPortRange": "3389",
+      "access": "Allow",
+      "direction": "Inbound",
+      "sourcePortRange": "*",
+      "destinationAddressPrefix": "*"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines",
+      "apiVersion": "2018-06-01",
+      "name": "[concat(parameters('prefix'),'DC')]",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Network/networkInterfaces/', toLower(parameters('prefix')), '-ni-DC')]"
+      ],
+      "properties": {
+        "osProfile": {
+          "computerName": "[concat(parameters('prefix'),'DC')]",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]",
+          "windowsConfiguration": {
+            "provisionVmAgent": "true"
+          }
+        },
+        "hardwareProfile": {
+          "vmSize": "[variables('generalSettings').virtualMachinessize]"
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "MicrosoftWindowsServer",
+            "offer": "WindowsServer",
+            "sku": "2016-Datacenter",
+            "version": "latest"
+          },
+          "osDisk": {
+            "osType": "Windows",
+            "name": "[concat(parameters('prefix'),'DC-OsDisk')]",
+            "createOption": "FromImage",
+            "caching": "ReadWrite",
+            "managedDisk": {
+              "storageAccountType": "[variables('generalSettings').storageAccountType]"
+            },
+            "diskSizeGB": 150
+          },
+          "dataDisks": []
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces', concat(toLower(parameters('prefix')), '-ni-DC'))]"
+            }
+          ]
+        },
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": false
+          }
+        }
+      }
+    },
+    {
+      "name": "[concat(parameters('prefix'),'DC/WorkFlow')]",
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "apiVersion": "2018-06-01",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/',parameters('prefix'),'DC')]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.Powershell",
+        "type": "DSC",
+        "typeHandlerVersion": "2.21",
+        "autoUpgradeMinorVersion": true,
+        "settings": {
+          "ModulesUrl": "https://raw.githubusercontent.com/yizhongwu/azure-quickstart-templates/master/sccm-currentbranch/dsc/CreateADRootDC.ps1.zip",
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

Reply:
If I use _ArtifcatLocation in prereq template, it reports 404 (file not found) error in Travis CI. From the error log, it looks trying to get the file from https://raw.githubusercontent.com/**Azure**/azure-quickstart-template/.... It isn't trying to find it under my fork. So I changed to the hardcoded path. It passed. So I think I will change it back to _ArtifcatLocation once it merges to the main once.

Reply:
If you author this according to the BP guide it will work...